Tasks/AzureFileCopy/Utility.ps1

```diff
@@ -984,190 +928,209 @@ function Get-AzureVMsCredentials
+            ThrowError -errorMessage (Get-VstsLocString -Key "AFC_CopyFailed" -ArgumentList $_)
         }
     }
+
+    Write-Verbose "Successfully finished sequential file copy"
 }
 
-function Copy-FilesParallellyToAzureVMs
+function Copy-FilesToAzureVMsFromStorageContainer
 {
-    param([string][Parameter(Mandatory=$true)]$storageAccountName,
-          [string][Parameter(Mandatory=$true)]$containerName,
-          [string][Parameter(Mandatory=$true)]$containerSasToken,
-		  [string]$blobStorageEndpoint,
-          [string][Parameter(Mandatory=$true)]$targetPath,
-          [string][Parameter(Mandatory=$true)]$azCopyLocation,
-          [object][Parameter(Mandatory=$true)]$azureVMResourcesProperties,
-          [object][Parameter(Mandatory=$true)]$azureVMsCredentials,
-          [string][Parameter(Mandatory=$true)]$cleanTargetBeforeCopy,
-          [string]$communicationProtocol,
-          [string]$skipCACheckOption,
-          [string][Parameter(Mandatory=$true)]$enableDetailedLoggingString,
-          [string]$additionalArguments,
-          [string][Parameter(Mandatory=$true)]$connectionType)
-
-    [hashtable]$Jobs = @{}
-    $dtlsdkErrors = @()
-    $deploymentUtilitiesLocation = Get-DeploymentModulePath
-    foreach ($resource in $azureVMResourcesProperties.Keys)
-    {
-        $resourceProperties = $azureVMResourcesProperties[$resource]
-        $resourceFQDN = $resourceProperties.fqdn
-        $resourceName = $resourceProperties.Name
-        $resourceWinRMHttpsPort = $resourceProperties.winRMHttpsPort
+    param(
+        [string[]]$targetMachineNames,
+        [pscredential]$credential,
+        [string]$protocol,
+        [object]$sessionOption,
+        [string]$blobStorageEndpoint,
+        [string]$containerName,
+        [string]$containerSasToken,
+        [string]$targetPath,
+        [bool]$cleanTargetBeforeCopy,
+        [bool]$copyFilesInParallel,
+        [string]$additionalArguments,
+        [string]$azCopyToolLocation
+    )
 
-        Write-Output (Get-VstsLocString -Key "AFC_CopyStarted" -ArgumentList $resourceName)
+    # Generate storage container URL
+    $containerURL = [string]::Format("{0}/{1}", $blobStorageEndpoint.Trim("/"), $containerName)
 
-        $job = Start-Job -ScriptBlock $AzureFileCopyJob -ArgumentList `
-                   $deploymentutilitieslocation, $resourceFQDN, $storageAccountName, $containerName, $containerSasToken, $blobStorageEndpoint, $azCopyLocation, $targetPath, $azureVmsCredentials, `
-                   $cleanTargetBeforeCopy, $resourceWinRMHttpsPort, $communicationProtocol, $skipCACheckOption, $enableDetailedLoggingString, $additionalArguments
+    $azCopyToolFileNames = Get-ChildItem $azCopyToolLocation | Select-Object -ExpandProperty Name
 
-        $Jobs.Add($job.Id, $resourceProperties)
+    $azCopyToolFileContents = @()
+    
+    foreach ($file in $azCopyToolFileNames)
+    {
+        $fullPath = Join-Path -Path $azCopyToolLocation -ChildPath $file
```

Comment:
You can skip out on doing this if you expand the property 'fullname' instead of 'name'

Reply:
done